### PR TITLE
[RFC] Tests around reported cases over DDC-2524

### PR DESCRIPTION
**Pull request by guilhermeblanco**
Fixes #707

### CODE REVIEW

**Comment by guilhermeblanco** [comment]
@mnapoli so far I fixed the `testSingle`. I'm thinking `testMany` might not be a problem of CommitOrderCalculator anymore, but rather a Persister issue.

File: lib/Doctrine/ORM/Internal/CommitOrderCalculator.php [hunk 1]
```diff
@@ -20,137 +21,163 @@
...
     /**
-     * @param \Doctrine\ORM\Mapping\ClassMetadata $class
+     * Visit a given node definition for reordering.
      *
-     * @return void
+     * {@internal Highly performance-sensitive method.}
+     *
+     * @param \stdClass $vertex
      */
-    public function addClass($class)
```
Comment (stof) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: removing public properties is a BC break, which must not be done in minor releases. You broke doctrine/data-fixtures. Please add the methods back

Comment (Ocramius) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: Agree here - the merge has to be reverted and delayed to 3.x :-(

Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: This class was never meant to be used by anyone else, not even other Doctrine projects.
`data-fixtures` should copy the new class and use on its own

Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: Here is a more detailed reason why I think this should not be reverted and why `data-fixtures` usage of the old class here is broken: https://github.com/doctrine/data-fixtures/pull/212#issuecomment-161023062

Comment (fsevestre) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: A solution as been decided for this BC break ? We temporarily fix the Doctrine ORM version to 2.5.2 on our project.

Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: @fsevestre yes, we'll not revert this commit. Explanation is available at data-fixtures comment I left in previous message.

Comment (colinodell) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: @guilhermeblanco Does that mean this fix will land in the next 2.5.x release?

I've got a different issue with commit ordering that this PR solves, so I really need it soon.

Comment (colinodell) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: Upon further review, it looks like my issue is likely related to cyclic dependencies, which explains why this fix would solve my issue.

Comment (guilhermeblanco) [inline lib/Doctrine/ORM/Internal/CommitOrderCalculator.php]: It'll only land in 2.6, as this introduces a BC break for data-fixtures project.

**Pull request merged**